src/main/java/com/conveyal/r5/analyst/progress/Task.java

```diff
@@ -50,66 +45,63 @@
 
     private int logFrequency = 0;
 
-    public String description;
+    public int totalWorkUnits = 1;
 
-    public int totalWorkUnits;
+    public int currentWorkUnit = 0;
 
-    public int currentWorkUnit;
+    public final Map<String, String> tags = new HashMap<>();
```

Comment:
Instead of this style we could `extend Task` for each specific case usage we have. For example, in Tasks with a user associated with it we need to store the `email` and `accessGroup` (possibly with a `UserTask`). In the `BundleController` we need to store the `bundleId`, `regionId`, which would need to `extend UserTask`. 

This is more "flexible" but maybe that is not needed. 

Part of the motivation here is that the `r5` package should not need to know anything about `UserPermissions`. But that it would be convenient for us to tag tasks that were sent to r5 with that data.

Reply:
Again I think we may have differing assumptions here. I don't really understand the use case for this flexibility, or the additional fields and tags. I didn't intend Tasks to store any detailed information like bundle IDs, just represent and track background task progress for display in the UI. Any state specific to the work being done was intended to be encapsulated in the TaskAction implementation.

Reply:
Yes, and displaying in the UI requires a degree of filtering. 
- We were already intending to do that by `email`/`accessGroup`.  
- `regionId` also seems desirable. 
- Then possibly a `category` or `page` filter. For example: "show me all active regional analyses"

`bundleId` may not be necessary, especially if we intend to skip saving to MongoDB until completion. We could return the specific `taskId` from the `BundleController.create` method in order to track it.

Reply:
I'm quite confused by that statement. I thought `TaskAction` was meant to be just a single function?